scalacache/src/main/scala-3/com/dwolla/tagless/scalacache/package.scala

```diff
@@ -0,0 +1,140 @@
+package com.dwolla.tagless.scalacache
+
+import _root_.scalacache.*
+import cats.*
+import cats.tagless.*
+import cats.tagless.aop.*
+import com.dwolla.tracing.*
+import io.circe.Codec
+import io.circe.generic.semiauto.*
+import natchez.*
+import LowPriorityTraceableValueInstances.*
+
+import scala.concurrent.duration.Duration
+
+/**
+ * An `InvariantK[Cache[*[_], K, V]]` instance for arbitrary `K` and
+ * `V` types. This must be `InvariantK` and not e.g. `FunctorK`
+ * because of the `F[V]` parameter on the `cachingF` method.
+ */
+given[K, V]: InvariantK[[F[_]] =>> Cache[F, K, V]] =
```

Comment:
I couldn't figure out how to write this type lambda `[F[_]] =>> Cache[F, K, V]` in a way that is compatible with Scala 2 and Scala 3, which is why I duplicated this file in the `scala-2` and `scala-3` source directories. We may be able to collapse them back into a single file if we can figure that out, but I think this is a good workaround for now.